### ISSUE TITLE
Don't work with Playwright 1.38

### ISSUE THREAD

**Issue by aninde** [lost-pixel/lost-pixel#327]
### Bug description

In my project I'm using 
    "@playwright/browser-chromium": "^1.39.0",
    "@playwright/browser-firefox": "^1.38.0",
    "@playwright/browser-webkit": "^1.38.0",
    "@playwright/test": "^1.38.0"

When I tried to run Lost Pixel I received
❌ browserType.launch: Executable doesn't exist at. ../ms-playwright/chromium-1076/chrome-mac/Chromium.app/Contents/MacOS/Chromium

Suggested  `npx playwright install` didn't help, but I was able to fix it with workaround
in the folder `/ms-playwright/` I copied existing `/ms-playwright/chromium-1080/`, paste it and rename it to chromium-1076 

I bet upgrading Playwright to 1.38 on you package.json should solve this globally. 

### How to reproduce

1. Use lost-pixel v3.8.0 with Playwright 1.38
2. Build storybook
3. Run Lost pixel
4. See error in Terminal

**Comment by d-ivashchuk** [comment]
Thanks for reporting, we will take a look into this one! If you would have any other questions regarding lost pixel - feel free to ping me anytime 🙌🏼

**Comment by Hai-San** [comment]
Same problem here with playwright 1.40.0

In my case the folder name is `chromium-1091` not `chromium-1076`

**Comment by greenskybluephish** [comment]
Seems like this is caused by listing `playwright` as a [dev dependency](https://github.com/lost-pixel/lost-pixel/blob/3b7fc699469a00f3e34cc367f2d8bc58c8da3b5b/package.json#L79), but the build is expecting the browsers that were used with the 1.37 version.

**Comment by d-ivashchuk** [comment]
I am sorry for not providing enough attention to this! will try to reproduce it once more. @greenskybluephish as you already got some notion on what's going on, do you want to give it a run as a PR fix?

**Comment by chbiel** [comment]
Experiencing the same, reproducable with:

```
mkdir lost-pixel-test
cd lost-pixel-text
npm init
npm i --save-dev lost-pixel @playwright/test
npx lost-pixel init-ts
npx lost-pixel
```
Results in error
`browserType.launch: Executable doesn't exist at /home/dejhcb0c/.cache/ms-playwright/chromium-1076/chrome-linux/chrome`

**Comment by d-ivashchuk** [comment]
@chbiel if you wanna give fix a go - very much welcome. Otherwise we will tackle it over the weekend!

**Comment by chbiel** [comment]
@d-ivashchuk I tried, but following the contribution guide does also not work right now :/

**Comment by d-ivashchuk** [comment]
would love to assist you with any troubles(and improve the docs for contribution guide at the same time :D) what challenges did you face?

**Comment by chriskalmar** [comment]
I'm very sorry for the late reply.
For the Lost Pixel Github Action to work correctly, we need to have a fixed version of Playwright.
Still, I believe, by going with a peerDependency we could solve that problem for local usages.

I've created a tiny beta npm version to test that theory.
If you (@aninde @Hai-San @chbiel) could give it a shot and report back 🙏  if it solves the issue it would be a great help and we could ship it as a regular version right away.

This is the version: `3.8.3-beta.1`

```
npm i lost-pixel@3.8.3-beta.1
```

Of course, this requires you to also install your own `playwright-core`, but that's anyway what you wanted in the first place.
I hope it works 🤞

**Comment by Maxim-Mazurok** [comment]
Hey guys, had same issue, found a solution I think.

Basically when you run `npx playwright` - it will use latest version, not the one that is installed as lost-pixel's dependency.

And in the latest version it has different browsers.json file with default install version of chrome.

So the most sustainable solution is to run `./node_modules/lost-pixel/node_modules/.bin/playwright-core install --with-deps`

Otherwise you can modify you command to use `npx playwright@1.37.0` but you need to make sure version matches one from lost-pixel deps.

I guess alternatively you could configure your own browsers.json I'm not sure, first time using playwright `node_modules/lost-pixel/node_modules/playwright-core/browsers.json`
Hope this helps, cheers!

**Comment by chriskalmar** [comment]
Yes, @Maxim-Mazurok, this solution works too, but there's a drawback:
It binds your project to the Playwright version of the currently used Lost Pixel version.

Of course, some projects use Playwright just via Lost Pixel and that's fine, but if a project wants to run Playwright in a new version (to e.g. create integration tests) it would become a problem. That's why I moved `playwright-core` to `peerDependencies` to allow users to pick their own Playwright version.

Maybe you could give `lost-pixel@3.8.3-beta.1` a shot and confirm that it works for you as well @Maxim-Mazurok ?

**Comment by Maxim-Mazurok** [comment]
Ah I see, makes sense, didn't read thread in detail. Sorry, don't have bandwidth for now, also not using playwright for anything other than lost-pixel for now, so happy with my setup, will be happy to try if anything changes and will report back, cheers!

**Comment by dmitrc** [comment]
It seems like there is still an issue here, or am I doing something wrong?

In my `package.json` I have:
```
"@playwright/test": "^1.42.1",
"playwright-core": "^1.42.1", // just in case
```

When running `lost-pixel`, I get:

```
❌ browserType.launch: Executable doesn't exist at C:\Users\UserName\AppData\Local\ms-playwright\chromium-1091\chrome-win\chrome.exe
╔═════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright Test or Playwright was just installed or updated. ║
║ Please run the following command to download new browsers:              ║
║                                                                         ║
║     npx playwright install                                              ║
║                                                                         ║
║ <3 Playwright Team                                                      ║
╚═════════════════════════════════════════════════════════════════════════╝
```

When checking the browsers directory, I see a newer `chromium-1097` instead.

**Comment by kraegpoeth** [comment]
@d-ivashchuk I have the samme issue as @dmitrc. I need to run 1.42.1 for out e2e and integration tests, but lost-pixel only works with chromium-1091, which is a dep of playwright 1.40.1. Essentially i can choose between having my snapshots working or my e2e tests :/ 
I not quite sure i understood your potential workaround with `peerDependency` @chriskalmar ?
Should i just move @playwright/test from devDep to peerDep?

**Comment by okonet** [comment]
Same problem here. I installed `npm i lost-pixel@3.8.3-beta.1` but the problem still remains.

```
npx lost-pixel local
Version: 3.8.3-beta.1
Loading project config ...
...
Removing 0 files from .lostpixel/current/
Removing 0 files from .lostpixel/difference/
Found 0 pages from external loader

=== [Page Mode] http://localhost:3000/ ===

Prepared 13 pages for screenshots on chromium
❌ browserType.launch: Executable doesn't exist at /Users/okonet/Library/Caches/ms-playwright/chromium-1076/chrome-mac/Chromium.app/Contents/MacOS/Chromium
╔═════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright Test or Playwright was just installed or updated. ║
║ Please run the following command to download new browsers:              ║
║                                                                         ║
║     npx playwright install                                              ║
║                                                                         ║
║ <3 Playwright Team                                                      ║
╚═════════════════════════════════════════════════════════════════════════╝
Sending anonymized telemetry data.
```

**Comment by tomekbuszewski** [comment]
I am facing the same issue. It seems that Lost Pixel looks for a particular version of a browser, rather than for the one Playwright ships.

I've tried circumventing this by installing browser locally (with `PLAYWRIGHT_BROWSERS_PATH=0 npx playwright install`) and pointing Lost Pixel to it via config's `browserLaunchOptions.chromium.executablePath`, but I got EACCES error:

```
browserType.launch: Failed to launch: Error: spawn node_modules/playwright/node_modules/playwright-core/.local-browsers/chromium-1140 EACCES
```

Any help would be appreciated!

**Comment by brettdorrans** [comment]
I am also facing the same issue. None of the previous workarounds in this thread seem to work given changes since those versions.

This is on a clean install of an empty application following the storybook example in the docs: https://github.com/lost-pixel/lost-pixel-examples/tree/main/example-storybook-v6.5-storystore-v7 

```
❌ browserType.launch: Executable doesn't exist at /home/brett/.cache/ms-playwright/chromium-1134/chrome-linux/chrome
```

Manually renaming this directory to the one lost-pixel expects works fine, so it's a matter of lost-pixel not resolving the correct version of chromium playwright is installing.

@d-ivashchuk if you are able to point me towards the code that handles this, I may be able to provide a fix PR.

**Comment by Kanti** [comment]
Downgrading `@playwright/test` to version `1.47.2` helped me.

**Comment by Robind-ev** [comment]
I have problems with this as well.

Playwright installs chromium 1154, Lost pixel expects 1134.

Its no longer just a question of re-naming the folder, as the headless that is executed is no longer supported in chrome.

[pid=1505690][err] Old Headless mode has been removed from the Chrome binary. Please use the new Headless mode (https://developer.chrome.com/docs/chromium/new-headless) or the chrome-headless-shell which is a standalone implementation of the old Headless mode (https://developer.chrome.com/blog/chrome-headless-shell).

**Comment by slamer59** [comment]
Hello, I am facing the same isue. any workaround ?

**Comment by karol-bujacek** [comment]
The problem is still actual. Downgrade playwright version helps to run lost-pixel, but it is not ideal.
